Banyan/src/clusters.jl

```diff
@@ -165,20 +177,43 @@ function get_clusters(; kwargs...)
             c["s3_read_write_resource"],
         ) for (name, c) in response["clusters"]
     )
+
+    # Cache info
+    global clusters
+    for (name, c) in clusters_dict
+        clusters[name] = c
+    end
+
+    return clusters_dict
+end
+
+function get_cluster_s3_bucket_arn(cluster_name=get_cluster_name(); kwargs...)
+    configure(; kwargs...)
+    global clusters
+    # Check if cached, sine this property is immutable
+    if !haskey(clusters, cluster_name)
+        get_cluster(cluster_name)
+    end
+    return clusters[cluster_name].s3_bucket_arn
 end
 
 function get_cluster_s3_bucket_name(cluster_name=get_cluster_name(); kwargs...)
     configure(; kwargs...)
-    cluster = get_cluster(cluster_name)
-    return s3_bucket_arn_to_name(cluster.s3_bucket_arn)
+    return s3_bucket_arn_to_name(get_cluster_s3_bucket_arn(cluster_name))
 end
 
-get_cluster(name::String=get_cluster_name(), kwargs...) = get_clusters(; kwargs...)[name]
+get_cluster(name::String=get_cluster_name(), kwargs...) = get_clusters(name; kwargs...)[name]
 
 get_running_clusters(args...; kwargs...) = filter(entry -> entry[2].status == :running, get_clusters(args...; kwargs...))
 
 function get_cluster_status(name::String=get_cluster_name(), kwargs...)
-    c = get_clusters(; kwargs...)[name]
+    global clusters
+    if !haskey(clusters, name)
+        if clusters[name].status == :failed
+            @error c.status_explanation
+        end
+    end
```

Comment:
This doesn't look right. We shouldn't be trying to access `clusters[name]` if `!haskey(clusters, name)`.

Reply:
Oh, thanks for catching this.

Reply:
Fixed by https://github.com/banyan-team/banyan-julia/pull/78/commits/76d8e6970961e50e9e5117be82807de6c5279ca8